### ISSUE TITLE
Add markdown support

### ISSUE THREAD

**Issue by ifiokjr** [remirror/remirror#553]
### Description

Add a block node that only displays markdown text content. It is similar to the `CodeBlockExtension` since it uses refractor to automatically decorate matching markdown tokens. However, it will not be wrapped in a code block. Since refractor is supported we should also automatically have support for syntax highlighting within code fences.

Once ready this will be the recommended way to create markdown editors.

```ts
import { DocExtension } from 'remirror/extension/doc';
import { TextExtension } from 'remirror/extension/text';
import { MarkdownBlockExtension } from 'remirror/extension/markdown-block';
import { RemirrorManager } from 'remirror/core';

const manager = RemirrorManager.create(() => [
  new MarkdownBlockExtension(),
  new DocExtension({ content: 'markdownBlock+' }),
  new TextExtension(),
]);
```

Questions to answer:

- [ ] Will this work without being wrapped in a code tag?
- [ ] How should the enter key be handled?
  - Should it create a new `markdownBlock` or insert a break tag.

Items to work on.

- [x] Seek an interested sponsor.
- [x] Create basic `MarkdownBlockExtension`.
- [x] Verify it works in the playground.

**Comment by ajnatural** [comment]
Is it possible to pay to move this forward on the roadmap? Please let me know your email to discuss

**Comment by ifiokjr** [comment]
@ajnatural it's probably easiest to private message me via our discord https://remirror.io/chat.

In case that's not possible, email also works - `ifiok(at)kickjump(dot)co`.

**Comment by ifiokjr** [comment]
This is being worked on. Hopefully I'll have the new package available this week.

**Comment by ifiokjr** [comment]
This is available in the beta release #706.
